api/naming/naming.go

```diff
@@ -27,3 +28,17 @@ func Task(t *api.Task) string {
 }
 
 // TODO(stevvooe): Consolidate "Hostname" style validation here.
+
+// Runtime returns the runtime name from a given spec.
+func Runtime(t api.TaskSpec) string {
+	switch r := t.GetRuntime().(type) {
+	case *api.TaskSpec_Attachment:
+		return "attachment"
+	case *api.TaskSpec_Container:
+		return "container"
+	case *api.TaskSpec_Generic:
+		return strings.ToLower(r.Generic.Kind)
+	default:
+		return "none"
```

Comment:
It used to panic. But we have **so much** code in our unit tests that just pass a service (or task) with no runtime. I'm not sure it was worth the hassle of fixing it all.

Thoughts?